src/UiDatepicker.vue

```diff
@@ -199,6 +199,16 @@ export default {
     },
 
     computed: {
+        dateObject() {
+            let value = this.value
+
+            if (typeof value === 'string') {
+                value = new Date(value)
+            }
+
+            return value && !isNaN(value) ? dateUtils.clone(value) : null
```

Comment:
Why do we need the `isNaN` check here?

Also why clone the date object for return?